src/jp2image.cpp

```diff
@@ -339,18 +339,17 @@ namespace Exiv2
 
                             if (rawData.size_ > 0)
                             {
-                                // Find the position of Exif header in bytes array.
-                                long pos = (     (rawData.pData_[0]      == rawData.pData_[1])
-                                           &&    (rawData.pData_[0]=='I' || rawData.pData_[0]=='M')
-                                           )  ? 0 : -1;
-
-                                // #1242  Forgive having Exif\0\0 in rawData.pData_
-                                const byte exifHeader[] = { 0x45, 0x78, 0x69, 0x66, 0x00, 0x00 };
-                                for (long i=0 ; pos < 0 && i < rawData.size_-(long)sizeof(exifHeader) ; i++)
-                                {
-                                    if (memcmp(exifHeader, &rawData.pData_[i], sizeof(exifHeader)) == 0)
-                                    {
-                                        pos = i+sizeof(exifHeader);
+                                bool foundPos{ false };
+                                size_t pos{};
+                                if ((rawData.pData_[0] == rawData.pData_[1]) &&
```

Comment:
Please check whether `rawData.size_` is at least 2 before accessing the second element.

Reply:
Ok.

Reply:
I have a question: in line 346 you are setting `foundPos=true` but `pos` is not set. Looking at the previous code it seems that we were setting that value to `0` when we were entering that if block.

Although I know that `pos` is initialised to 0 by using the `{}`, I think it would be more clear to indicate the value explicitly:

```cpp
 size_t pos{0};
```

Reply:
OK, will do.